src/use_case/rooms/Response.java

```diff
@@ -0,0 +1,23 @@
+package use_case.rooms;
+
+public class Response<T> {
+  private String error; // Nullable field
+  private T val;
+
+  public Response(T val, String error) {
+    this.val = val;
+    this.error = error;
+  }
```

Comment:
```suggestion
  }
  public Response(T val) {
    this.val = val;
    this.error = null;
  }
```

Reply:
I think this is a good suggestion @alexrosen45.

Reply:
Yeah, I just implemented this in last commit - forgot to mention.